kernel/vars.ml

```diff
@@ -177,6 +194,9 @@ let subst_of_rel_context_instance sign l =
     | _ -> CErrors.anomaly (Pp.str "Instance and signature do not match.")
   in aux [] (List.rev sign) l
 
+let subst_of_rel_context_instance sign v =
```

Comment:
I think it's a bad idea to introduce this function with this name. It's performing a potentially costly conversion that should be avoided, while getting the short name without the `_vect` prefix. I definitely prefer the code as it was before, i.e. with the `Array.to_list` function calls in the clients instead. Now the default is the wrong one and there is no way to realize that the correct call should be the other one without looking at the code.

Reply:
On one side, it is by similarity with (the recent uniformization) about `Context.{Rel,Named}.instance{,_list}` which also gives a short name to the version with an array-to-list conversion.

On another side, wrt efficiency, why not then to rewrite `subst_of_rel_context_instance` directly on `array` since in practice the argument comes from an `App` node and is an array most of the time.

Similarly, `Context.{Rel,Named}.instance` could be rewritten directly on `array`??

At the level of terminology, we could also force a `_list`/`_vect` suffix so that there is no bias on using one over the other??

Reply:
Meh. That looks like a lot of work. Let's merge as-is for now, we'll refine after.

> in practice the argument comes from an App node

Yes and no. Most of the time we manipulate the array and take slices of it, and it could as well have been turned into a list upfront. But it's all a matter of style.